### PR TITLE
Merge crazycanvas, server, client and sandbox projects

### PR DESCRIPTION
**Pull request by TheoBerlin**
We still have separate project files, but they are identical besides from two things:
- Separate binaries (.exe files)
- Different debugargs, eg. `"--state=sandbox"`

Also, removed the SharedLib platform.

Each project has visual studio pass a unique flag to the application when debugging to decide which state to start with, and which engine config to load. The server, client and sandbox states don't create a menu when starting up.

### CODE REVIEW

File: CrazyCanvas/Source/CrazyCanvas.cpp [hunk 1]
```diff
@@ -28,11 +30,31 @@ CrazyCanvas::CrazyCanvas(const argh::parser& flagParser)
...
+		pStartingState = DBG_NEW NetworkingState();
+	}
+	else if (stateStr == "server")
+	{
+		pStartingState = DBG_NEW ServerState();
+	}
+	else if (stateStr == "benchmark")
+	{
+		pStartingState = DBG_NEW BenchmarkState();
+	}
```
Comment (frozenbanana) [inline CrazyCanvas/Source/CrazyCanvas.cpp:57]: We need a case for Main menu as well.

File: LambdaEngine/Source/Engine/EngineConfig.cpp [hunk 2]
```diff
@@ -31,9 +37,7 @@ namespace LambdaEngine
 
 		s_ConfigDocument.ParseStream(inputStream);
 
-		fclose(pFile);
-
-		return true;
+		return fclose(pFile) == 0;
 	}
 
 	bool EngineConfig::WriteToFile()
```
Comment (frozenbanana) [inline LambdaEngine/Source/Engine/EngineConfig.cpp:43]: Would be safer if `EngineConfig::WriteToFile()` got updated as ReadToFile. File path for writing is still hardcoded.

**Review by frozenbanana — CHANGES_REQUESTED** [review]
See comment. Otherwise it looks good to me.

File: LambdaEngine/Source/Engine/EngineConfig.cpp (hunk 2 above)
Comment (TheoBerlin) [reply LambdaEngine/Source/Engine/EngineConfig.cpp:43]: Good spot, fixed.

**Review by skumbananaaa — APPROVED** [review]
Looks good

**Review by TheRealChriAnd — APPROVED** [review]
Looking good!

**Review by frozenbanana — APPROVED** [review]
Looks good!

**Pull request merged**